charts/canvas-vault/configmap/canvas-vault-certificate.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  # {{ .Release.Name }}-vault-tls can not be used as name, because the extra env variables for hc vault 
+  # are defined in values yaml, which does not allow templating 
+  name: canvasvault-tls
+  # TODO[FH]: this should be
+  # namespace: {{ include "canvasvault.namespace" . }}
```

Comment:
Unfortunately the custom resource "Certificate" is not known at deploy time, because Cert-Manager is deployed in parallel. So I made a workaround to not directly deploy the certificate, instead of a ConfigMap is created which contains the Certificate to deploy and the deployment is done using kubectl in the post-install hook.
So, the templating does not work for files that was the reason, why I hardcoded the namespace.

In the Clinic Session @LesterThomas and @brian-burton already talked about splitting the all-at-once installation into two or event three deployments. Then Cert-Manager would already be present before installing HashiCorp Vault and the Certificate can be deployed as a normal helm template.

At the moment I would leave it this way.